Sources/Datadog/Utils/SwiftExtensions.swift

```diff
@@ -13,3 +13,11 @@ extension Optional {
         }
     }
 }
+
+extension Date {
+    // NOTE: RUMM-182 counterpart of currentTimeMillis in Java
+    // https://docs.oracle.com/javase/7/docs/api/java/lang/System.html#currentTimeMillis()
+    var currentTimeMillis: UInt {
+        return UInt(self.timeIntervalSince1970 * 1_000)
```

Comment:
why  "<" than UInt64.max and not "<=" ?

Reply:
Hm, no reason I think, we just overlooked this 🤔.

Reply:
this solution does not cover all the cases.

`fileCreationDate.timeIntervalSinceReferenceDate * 1_000` may overflow as well

Reply:
@bmermet yes, and that's exactly why we check for overflow in `FilesOrchestrator`. We can reuse that code and create extension for safe conversion:

<img width="696" alt="Screenshot 2020-04-07 at 18 22 19" src="https://user-images.githubusercontent.com/2358722/78694303-bdd27700-78fc-11ea-8ed1-b8f57afdedc6.png">

Reply:
This code above crashes if `double` is close to `Double.greatestFiniteMagnitude`

That one below seems like a better solution
```swift
extension FixedWidthInteger {
    init(fromDoubleWithReportingOverflow double: Double) throws {
        guard let converted = Self(exactly: double.rounded()) else {
            throw FixedWidthIntegerError.overflow(overflowingDouble: double)
        }
        self = converted
    }
}

enum FixedWidthIntegerError: Error {
    case overflow(overflowingDouble: Double)
}
```

1. this extension works for all integer types
2. it doesn't contain confusing logic (eg: <UInt64.min, >=UInt32.max, etc.)
3. `init(exactly:)` is a failable initializer, so it doesn't throw or crash

Reply:
What does it mean "if `double` is close to `Double.greatestFiniteMagnitude`" 🤔?

Reply:
I haven't calculated the exact range yet but if `double` is between little more than `0.9999999999999999` and `1.0` times of `Double.greatestFiniteMagnitude`, it **crashes**.

[`init(exactly:)`](https://github.com/apple/swift/blob/master/stdlib/public/core/Integers.swift#L3080) uses the same method with [`init(:)`](https://github.com/apple/swift/blob/master/stdlib/public/core/Integers.swift#L3054) internally

Reply:
`Double.greatestFiniteMagnitude` is an abstraction over "infinity", so it shouldn't be used for arithmetic operations - this is IMHO asking for trouble 💥.

Whatever works is OK for me 👍. However, the plain `double >= UInt64.min && double  <= UInt64.max` is way more readable (also intuitive and proven on production) than `FixedWidthInteger(exactly: double.rounded())`.

Reply:
Yes, I agree it is not intuitive, unfortunately it took me quite a while to find that method.

Here is my pros&cons:

### Pros:

1. We don't need to do sanity-check (ie: `double >= UInt64.min`), Swift does that for us 🎉 
    * It has less code and doesn't add extra logic
2. It never crashes, it works at even corner cases such as `Double.greatestFiniteMagnitude`
3. It has the same level of correctness with `init(:)` as they use the same internal method, the only difference is that this doesn't do `fatalError()`
    * I left URLs to their source code in the comment above ☝️ 

### Cons:

1. `init(exactly:)` is not a method name that you think of first when you are dealing with overflows
    * however, if we keep it in an extension and provide a better method name, it should be fine

Reply:
IMO ....I don't think you need any type safety check here as you are trying to convert a Date Double which will always be positive and smaller than Uint64.max (because Double is using his MSB for sign). If you are sure that this Date can never be negative you can just do the conversion directly ....if not you can just check if is ">=0" only.

Reply:
Ok nevermind what I said there....I definitely forgot how a Double is being represented. You definitely need that check there.

Reply:
### Safety check

_TL;DR: If we do boundry/safety check here, it will be nothing more than duplicating Swift's internal checks._

Swift does the exact same safety checks internally anyway.
The difference is, in case of failure:
1. `init(:)` calls `fatalError` (this is different than `error` or `exception`, the app just crashes) ❌ 
2. `init(exactly:)` returns `nil` ✅

Reply:
In this case I would go with the ">=,<" solution as it is more readable for me.